pybind11/setup_helpers.py

```diff
@@ -309,6 +309,9 @@ class ParallelCompile(object):
     envvar: Set an environment variable to control the compilation threads, like NPY_NUM_BUILD_JOBS
     default: 0 will automatically multithread, or 1 will only multithread if the envvar is set.
     max: The limit for automatic multithreading if non-zero
+    lazy: Do not try to recompile up-to-date outputs if True
```

Comment:
Wondering if "lazy" is the right thing here. When I hear lazy in computational context, I think of postponing work until it's really necessary, rather than not doing work that has already be done. But maybe I'm confusing the general "lazy" and specific "lazy evaluation"?

Reply:
Feel free to suggest a better term. I has `cached` originally.

Reply:
Any reason you discarded `cached`? Otherwise, `use_cache`, perhaps?

But no change needed for just me. Just mentioning and seeing if you or anyone else would feel the same about `lazy`.

Reply:
`cached=True` doesn't really explain what it's doing that well, I though. How about `if/as_needed=True`?

Reply:
True, yes. Or `always_recompile=False`?

Reply:
> True, yes. Or `always_recompile=False`?

Nah, that's not fully clear either.
So, OK, from that perspective, `lazy` is a great term, it just has an unfortunate connotation.

Reply:
`incremental=True`?

Reply:
I actually liked `always_recompile=False` better than `incremental=True`, which sounds like incremental linking in Windows.

How about `only_changed=True`?

Reply:
(I was thinking of incremental builds, but OK)

Could also work, yes. None of these all fully convinced me, so far. But none of them are awful either, so ...